torch/_dynamo/side_effects.py

```diff
@@ -430,6 +434,13 @@ def mutation(self, var):
     def _get_modified_vars(self):
         return [var for var in self.id_to_variable.values() if self.is_modified(var)]
 
+    def get_new_function(self, var):
+        if isinstance(var, variables.UserDefinedDictVariable):
+            return "dict_new"
+        elif isinstance(var, variables.UserDefinedTupleVariable):
+            return "tuple_new"
+        return "object_new"
```

Comment:
Might be cleaner as a method (you override) on VariableTracker.

Reply:
Will address in a follow up PR where I am adding UserDefinedList support.